src/lib/quoter.js

```diff
@@ -52,9 +52,14 @@ class Quoter {
     const hop = this._findBestHopForAmount(request)
     if (!hop) return null
 
-    // If we know a local route to the destinationAccount, use the local route.
+    // note the confusion between `next`, `destination`, and `final`;
+    // `finalAmount` will be renamed to `destinationAmount` later in
+    // the hopToQuote function.
+    if (hop.sourceAmount && hop.finalAmount) {
```

Comment:
This line only checks that the routing table lookup was successful. If you were quoting by sourceAmount, and you found the finalAmount, you're done.  If you were quoting by finalAmount, and you found the sourceAmount, you're also done. 

Btw, we should really stop using destinationAmount in this way, can we rename it to sourceAmount ; next Amount ; finalAmount?